trace-callbacks.sh

```diff
@@ -0,0 +1,34 @@
+# Copyright (c) 2022 All rights reserved.
+# SPDX-License-Identifier: GPL-2.0-only
+# File: trace-callbacks.sh
+# Purpose: Trace every N'Siq CppStyle callback for a given input source file.  Using
+#          the trace, the rule developer can plan how to write a new rule.
+
+FILEFILTER=filefilter.trace-callbacks.txt
+
+usage () {
+    echo "USAGE: trace-callbacks.sh path-to-source-file"
+}
+
+delete_filefilter () {
+    rm $FILEFILTER
+}
+
+if [ $# -eq 0 ]; then
+    echo "No file path argument provided"
+    usage
+    exit 1
+fi
+
+if ! [[ "$(python3 -V)" =~ "Python 3" ]]; then
+    echo "python3 not found"
+    exit 2
+fi
```

Comment:
I did not fathom the point here.  What is "exe.py"?  Are you advocating creating an executable on Windows/Linux, rather than natively running "python"/"python3"?  Please elaborate on what you have in mind.

Reply:
I usually skip the nsiqcppstyle_ prefix. So, when i say exe.py, it's same as `nsiqcppstyle_exe.py`

And I'm advocating for replacing the sh and bat files with a python script which is able to re-use code from the exe.py file. Hope it makes more sense now

Reply:
As I wrote below:

> Nice idea - if the user can run the trace.py script, he obviously has Python 3 ;-). Will attempt to implement.

Reply:
Implemented as trace-callbacks.py, called like this:

```
python trace-callbacks.py "D:\Junk\NsiqCppStyle\issue_34\hello-world.cpp"
python trace-callbacks.py "D:\Junk\NsiqCppStyle\issue_34\hello-world.cpp" "D:\Junk\NsiqCppStyle\issue_34\hello-world-2.cpp"
```